documentation/API-GUIDELINES.md

```diff
@@ -15,20 +15,20 @@ The following paragraphs contain additional recommendations.
 - If a driver supports multiple peripheral instances (for example, I2C0 is one such instance):
-  - The peripheral instance type should be positioned as the last type parameter of the driver type.
-  - The peripheral instance type should default to a type that supports any of the peripheral instances.
-  - The author is encouraged to use `crate::any_peripheral` to define the "any" peripheral instance type.
-  - The driver should implement a `new` constructor that automatically converts the peripheral instance into the any type, and a `new_typed` that preserves the peripheral type.
+  - The peripheral instance type must be positioned as the last type parameter of the driver type.
+  - The peripheral instance type must default to a type that supports any of the peripheral instances.
+  - The author must to use `crate::any_peripheral` to define the "any" peripheral instance type.
+  - The driver must implement a `new` constructor that automatically converts the peripheral instance into the any type, and a `new_typed` that preserves the peripheral type.
 - If a driver only supports a single peripheral instance, no instance type parameter is necessary.
-- If a driver implements both blocking and async operations, or only implements blocking operations, but may support asynchronous ones in the future, the driver's type signature should include a `crate::Mode` type parameter.
-- By default, constructors should configure the driver for blocking mode. The driver should implement `into_async` (and a matching `into_blocking`) function that reconfigures the driver.
-  - `into_async` should configure the driver and/or the associated DMA channels. This most often means enabling an interrupt handler.
-  - `into_blocking` should undo the configuration done by `into_async`.
-- The asynchronous driver implemntation should also expose the blocking methods (except for interrupt related functions).
-- Consider adding a `Drop` implementation resetting the peripheral to idle state.
+- If a driver implements both blocking and async operations, or only implements blocking operations, but may support asynchronous ones in the future, the driver's type signature must include a `crate::Mode` type parameter.
+- By default, constructors must configure the driver for blocking mode. The driver must implement `into_async` (and a matching `into_blocking`) function that reconfigures the driver.
+  - `into_async` must configure the driver and/or the associated DMA channels. This most often means enabling an interrupt handler.
+  - `into_blocking` must undo the configuration done by `into_async`.
+- The asynchronous driver implemntation must also expose the blocking methods (except for interrupt related functions).
+- Drivers must have a `Drop` implementation resetting the peripheral to idle state.
```

Comment:
once #2544 lands we should advise to use the `PeripheralGuard` (probably something for that other PR after this one is merged)